pkg/cvo/metrics.go

```diff
@@ -83,6 +103,10 @@ version for 'cluster', or empty for 'initial'.
 			Name: "cluster_installer",
 			Help: "Reports info about the installation process and, if applicable, the install tool.",
 		}, []string{"type", "version", "invoker"}),
+		clusterVersionOperatorSuccessfulUpdateRetrieval: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "cluster_version_operator_successful_update_retrieval",
+			Help: "Reports when updates were last succesfully retrieved and, if currently not able to retrieve updates, reports reason.",
```

Comment:
> Reports when updates were last succesfully retrieved and, if currently not able to retrieve updates, reports reason.

If you really want this, I'd recommend to have 2 distinct metrics: one reporting the last update timestamp and another reporting the reason.

Reply:
@simonpasquier 

If I have two metrics, how do I configure my alert to pull from the two metrics? With the one metric I use $value and $labels.reason:

- alert: CannotRetrieveUpdates
      annotations:
        message: Cluster version operator has not retrieved updates since {{ "{{ $value | humanizeDuration }}" }}. Failure reason {{ "{{ $labels.reason }}" }}.

Reply:
Using a `reason` label and a timestamp value, as we have with 06c550ff9fed, makes sense to me.  If you still think it should be split into two metrics, can you explain your motivation?

Reply:
> can you explain your motivation?

I feel that conflating 2 meanings into metric (when was the last successful updates + what is the last failure reason) is a source of confusion. Also if the update fails for reason A then for reason B without updating the value of `cluster_version_operator_successful_update_retrieval` then the alert will reset and go back to `pending` because the `reason` label would have changed.

You can inject the reason in the annotation using the `"cluster_version_operator_reason_failure > 0" | query | first | label "reason"` construct (see https://prometheus.io/docs/prometheus/latest/configuration/template_reference/#queries).

Reply:
Added second metric as recommended.